Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -225,6 +223,12 @@ inline WICPixelFormatGUID PixelFormat() const {
         _impl.renderingIntent = intent;
         return *this;
     }
+
+    ~__CGImage() {
+        for (auto listener: _imageDestructionListeners) {
```

Comment:
It is the listener's job to make sure that it can be called on any thread.

Access to the listener list is not currently synchronized because there is ONE consumer.

In the future, _that consumer will be removed_.